adapters/http4s/src/main/scala/caliban/Http4sAdapter.scala

```diff
@@ -315,18 +325,22 @@ object Http4sAdapter {
       route(req.mapK(to)).mapK(from).map(_.mapK(from))
     }
 
-  private def wrapRoute[F[_]: Effect](route: HttpRoutes[Task])(implicit runtime: Runtime[Any]): HttpRoutes[F] = {
-    val toF: Task ~> F    = λ[Task ~> F](_.toIO.to[F])
-    val toTask: F ~> Task = λ[F ~> Task](_.toIO.to[Task])
+  private def wrapRoute[F[_]: Async](
+    route: HttpRoutes[Task]
+  )(implicit dispatcher: Dispatcher[F], runtime: Runtime[Any]): HttpRoutes[F] = {
```

Comment:
I would prefer passing a dispatcher explicitly.  
But in `zio-interop-cats` the dispatcher provided implicitly everywhere, so I'm not sure which side should we stay on.